File: clients/python/sdk/models/v1/tag.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import history_item

class Tag(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new Tag and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The following property
        self._following: Optional[bool] = None
        # The history property
        self._history: Optional[List[history_item.HistoryItem]] = None
        # The name property
        self._name: Optional[str] = None
        # The url property
        self._url: Optional[str] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Tag:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Tag
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return Tag()
    
    @property
    def following(self,) -> Optional[bool]:
        """
        Gets the following property value. The following property
        Returns: Optional[bool]
        """
        return self._following
    
    @following.setter
    def following(self,value: Optional[bool] = None) -> None:
        """
        Sets the following property value. The following property
        Args:
            value: Value to set for the following property.
        """
        self._following = value
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "following": lambda n : setattr(self, 'following', n.get_bool_value()),
            "history": lambda n : setattr(self, 'history', n.get_collection_of_object_values(history_item.HistoryItem)),
            "name": lambda n : setattr(self, 'name', n.get_str_value()),
            "url": lambda n : setattr(self, 'url', n.get_str_value()),
        }
        return fields
    
    @property
    def history(self,) -> Optional[List[history_item.HistoryItem]]:
        """
        Gets the history property value. The history property
        Returns: Optional[List[history_item.HistoryItem]]
        """
        return self._history
    
    @history.setter
    def history(self,value: Optional[List[history_item.HistoryItem]] = None) -> None:
        """
        Sets the history property value. The history property
        Args:
            value: Value to set for the history property.
        """
        self._history = value
    
    @property
    def name(self,) -> Optional[str]:
        """
        Gets the name property value. The name property
        Returns: Optional[str]
        """
        return self._name
    
    @name.setter
    def name(self,value: Optional[str] = None) -> None:
        """
        Sets the name property value. The name property
        Args:
            value: Value to set for the name property.
        """
        self._name = value
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_bool_value("following", self.following)
        writer.write_collection_of_object_values("history", self.history)
        writer.write_str_value("name", self.name)
        writer.write_str_value("url", self.url)
        writer.write_additional_data_value(self.additional_data)
    
    @property
    def url(self,) -> Optional[str]:
        """
        Gets the url property value. The url property
        Returns: Optional[str]
        """
        return self._url
    
    @url.setter
    def url(self,value: Optional[str] = None) -> None:
        """
        Sets the url property value. The url property
        Args:
            value: Value to set for the url property.
        """
        self._url = value
    

File: clients/node/sdk/models/v1/tag.ts
import {createHistoryItemFromDiscriminatorValue} from './createHistoryItemFromDiscriminatorValue';
import {HistoryItem} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Tag implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The following property */
    private _following?: boolean | undefined;
    /** The history property */
    private _history?: HistoryItem[] | undefined;
    /** The name property */
    private _name?: string | undefined;
    /** The url property */
    private _url?: string | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Instantiates a new Tag and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the following property value. The following property
     * @returns a boolean
     */
    public get following() {
        return this._following;
    };
    /**
     * Sets the following property value. The following property
     * @param value Value to set for the following property.
     */
    public set following(value: boolean | undefined) {
        this._following = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "following": n => { this.following = n.getBooleanValue(); },
            "history": n => { this.history = n.getCollectionOfObjectValues<HistoryItem>(createHistoryItemFromDiscriminatorValue); },
            "name": n => { this.name = n.getStringValue(); },
            "url": n => { this.url = n.getStringValue(); },
        };
    };
    /**
     * Gets the history property value. The history property
     * @returns a HistoryItem
     */
    public get history() {
        return this._history;
    };
    /**
     * Sets the history property value. The history property
     * @param value Value to set for the history property.
     */
    public set history(value: HistoryItem[] | undefined) {
        this._history = value;
    };
    /**
     * Gets the name property value. The name property
     * @returns a string
     */
    public get name() {
        return this._name;
    };
    /**
     * Sets the name property value. The name property
     * @param value Value to set for the name property.
     */
    public set name(value: string | undefined) {
        this._name = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeBooleanValue("following", this.following);
        writer.writeCollectionOfObjectValues<HistoryItem>("history", this.history);
        writer.writeStringValue("name", this.name);
        writer.writeStringValue("url", this.url);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the url property value. The url property
     * @returns a string
     */
    public get url() {
        return this._url;
    };
    /**
     * Sets the url property value. The url property
     * @param value Value to set for the url property.
     */
    public set url(value: string | undefined) {
        this._url = value;
    };
}
